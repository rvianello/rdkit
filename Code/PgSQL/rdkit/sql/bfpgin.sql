CREATE INDEX fpidx ON pgbfp USING gin (f);
CREATE INDEX maccsfpidx ON pgbfp USING gin (maccsf);

SET rdkit.tanimoto_threshold = 0.5;


SET enable_indexscan=off;
SET enable_bitmapscan=off;
SET enable_seqscan=on;

SELECT
    id, tanimoto_sml(rdkit_fp('CC(=NNC(=O)C1=CC(=C(C=C1)OC)OC)C2=CC=CC=N2'::mol), f) AS sml
FROM
	pgbfp
WHERE rdkit_fp('CC(=NNC(=O)C1=CC(=C(C=C1)OC)OC)C2=CC=CC=N2'::mol) % f
ORDER BY sml DESC, id limit 10;

SELECT
    id, tanimoto_sml(maccs_fp('CC(=NNC(=O)C1=CC(=C(C=C1)OC)OC)C2=CC=CC=N2'::mol), maccsf) AS sml
FROM
	pgbfp
WHERE maccs_fp('CC(=NNC(=O)C1=CC(=C(C=C1)OC)OC)C2=CC=CC=N2'::mol) % maccsf
ORDER BY sml DESC, id limit 10;


SET enable_indexscan=on;
SET enable_bitmapscan=on;
SET enable_seqscan=off;

SELECT
    id, tanimoto_sml(rdkit_fp('CC(=NNC(=O)C1=CC(=C(C=C1)OC)OC)C2=CC=CC=N2'::mol), f) AS sml
FROM
	pgbfp
WHERE rdkit_fp('CC(=NNC(=O)C1=CC(=C(C=C1)OC)OC)C2=CC=CC=N2'::mol) % f
ORDER BY sml DESC, id limit 10;

SELECT
    id, tanimoto_sml(maccs_fp('CC(=NNC(=O)C1=CC(=C(C=C1)OC)OC)C2=CC=CC=N2'::mol), maccsf) AS sml
FROM
	pgbfp
WHERE maccs_fp('CC(=NNC(=O)C1=CC(=C(C=C1)OC)OC)C2=CC=CC=N2'::mol) % maccsf
ORDER BY sml DESC, id limit 10;

SET enable_indexscan=on;
SET enable_bitmapscan=on;
SET enable_seqscan=on;

SELECT
    id, tanimoto_sml(rdkit_fp('CC(=NNC(=O)C1=CC(=C(C=C1)OC)OC)C2=CC=CC=N2'::mol), f) AS sml
FROM
	pgbfp
WHERE rdkit_fp('CC(=NNC(=O)C1=CC(=C(C=C1)OC)OC)C2=CC=CC=N2'::mol) % f
ORDER BY sml DESC, id limit 10;

SELECT
    id, tanimoto_sml(maccs_fp('CC(=NNC(=O)C1=CC(=C(C=C1)OC)OC)C2=CC=CC=N2'::mol), maccsf) AS sml
FROM
	pgbfp
WHERE maccs_fp('CC(=NNC(=O)C1=CC(=C(C=C1)OC)OC)C2=CC=CC=N2'::mol) % maccsf
ORDER BY sml DESC, id limit 10;

DROP INDEX fpidx;
DROP INDEX maccsfpidx;
