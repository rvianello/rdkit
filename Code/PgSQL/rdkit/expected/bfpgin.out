SET extra_float_digits=0;
CREATE INDEX fpidx ON pgbfp USING gin (f);
CREATE INDEX maccsfpidx ON pgbfp USING gin (maccsf);
SET rdkit.tanimoto_threshold = 0.5;
SET enable_indexscan=off;
SET enable_bitmapscan=off;
SET enable_seqscan=on;
SELECT
    id, tanimoto_sml(rdkit_fp('CC(=NNC(=O)C1=CC(=C(C=C1)OC)OC)C2=CC=CC=N2'::mol), f) AS sml
FROM
	pgbfp
WHERE rdkit_fp('CC(=NNC(=O)C1=CC(=C(C=C1)OC)OC)C2=CC=CC=N2'::mol) % f
ORDER BY sml DESC, id limit 10;
   id    |        sml        
---------+-------------------
 5206420 |                 1
 3955259 |  0.79020979020979
 2841607 | 0.753056234718826
  571783 | 0.746341463414634
  771595 | 0.742729306487696
  872813 | 0.725118483412322
  872878 | 0.722100656455142
 1219325 | 0.720657276995305
  872831 |  0.71264367816092
  770089 | 0.711670480549199
(10 rows)

SELECT
    id, tanimoto_sml(maccs_fp('CC(=NNC(=O)C1=CC(=C(C=C1)OC)OC)C2=CC=CC=N2'::mol), maccsf) AS sml
FROM
	pgbfp
WHERE maccs_fp('CC(=NNC(=O)C1=CC(=C(C=C1)OC)OC)C2=CC=CC=N2'::mol) % maccsf
ORDER BY sml DESC, id limit 10;
   id    |        sml        
---------+-------------------
 5206420 |                 1
  840468 |  0.86046511627907
 1954933 |              0.84
 3955259 | 0.791666666666667
 5224323 | 0.764705882352941
  498250 |              0.75
  771595 |              0.74
 5718132 |  0.73469387755102
 5638671 | 0.714285714285714
  872878 | 0.711538461538462
(10 rows)

SET enable_indexscan=on;
SET enable_bitmapscan=on;
SET enable_seqscan=off;
SELECT
    id, tanimoto_sml(rdkit_fp('CC(=NNC(=O)C1=CC(=C(C=C1)OC)OC)C2=CC=CC=N2'::mol), f) AS sml
FROM
	pgbfp
WHERE rdkit_fp('CC(=NNC(=O)C1=CC(=C(C=C1)OC)OC)C2=CC=CC=N2'::mol) % f
ORDER BY sml DESC, id limit 10;
   id    |        sml        
---------+-------------------
 5206420 |                 1
 3955259 |  0.79020979020979
 2841607 | 0.753056234718826
  571783 | 0.746341463414634
  771595 | 0.742729306487696
  872813 | 0.725118483412322
  872878 | 0.722100656455142
 1219325 | 0.720657276995305
  872831 |  0.71264367816092
  770089 | 0.711670480549199
(10 rows)

SELECT
    id, tanimoto_sml(maccs_fp('CC(=NNC(=O)C1=CC(=C(C=C1)OC)OC)C2=CC=CC=N2'::mol), maccsf) AS sml
FROM
	pgbfp
WHERE maccs_fp('CC(=NNC(=O)C1=CC(=C(C=C1)OC)OC)C2=CC=CC=N2'::mol) % maccsf
ORDER BY sml DESC, id limit 10;
   id    |        sml        
---------+-------------------
 5206420 |                 1
  840468 |  0.86046511627907
 1954933 |              0.84
 3955259 | 0.791666666666667
 5224323 | 0.764705882352941
  498250 |              0.75
  771595 |              0.74
 5718132 |  0.73469387755102
 5638671 | 0.714285714285714
  872878 | 0.711538461538462
(10 rows)

SET enable_indexscan=on;
SET enable_bitmapscan=on;
SET enable_seqscan=on;
SELECT
    id, tanimoto_sml(rdkit_fp('CC(=NNC(=O)C1=CC(=C(C=C1)OC)OC)C2=CC=CC=N2'::mol), f) AS sml
FROM
	pgbfp
WHERE rdkit_fp('CC(=NNC(=O)C1=CC(=C(C=C1)OC)OC)C2=CC=CC=N2'::mol) % f
ORDER BY sml DESC, id limit 10;
   id    |        sml        
---------+-------------------
 5206420 |                 1
 3955259 |  0.79020979020979
 2841607 | 0.753056234718826
  571783 | 0.746341463414634
  771595 | 0.742729306487696
  872813 | 0.725118483412322
  872878 | 0.722100656455142
 1219325 | 0.720657276995305
  872831 |  0.71264367816092
  770089 | 0.711670480549199
(10 rows)

SELECT
    id, tanimoto_sml(maccs_fp('CC(=NNC(=O)C1=CC(=C(C=C1)OC)OC)C2=CC=CC=N2'::mol), maccsf) AS sml
FROM
	pgbfp
WHERE maccs_fp('CC(=NNC(=O)C1=CC(=C(C=C1)OC)OC)C2=CC=CC=N2'::mol) % maccsf
ORDER BY sml DESC, id limit 10;
   id    |        sml        
---------+-------------------
 5206420 |                 1
  840468 |  0.86046511627907
 1954933 |              0.84
 3955259 | 0.791666666666667
 5224323 | 0.764705882352941
  498250 |              0.75
  771595 |              0.74
 5718132 |  0.73469387755102
 5638671 | 0.714285714285714
  872878 | 0.711538461538462
(10 rows)

DROP INDEX fpidx;
DROP INDEX maccsfpidx;
